src/sgame/sg_bot_ai.cpp

```diff
@@ -421,6 +421,27 @@ AINodeStatus_t BotDecoratorReturn( gentity_t *self, AIGenericNode_t *node )
 	return status;
 }
 
+AINodeStatus_t BotDecoratorMapStatus( gentity_t *self, AIGenericNode_t *node )
+{
+	AIDecoratorNode_t *dec = ( AIDecoratorNode_t * ) node;
+
+	AINodeStatus_t result;
+	switch ( BotEvaluateNode( self, dec->child ) )
+	{
+	case STATUS_FAILURE:
+		result = ( AINodeStatus_t ) AIUnBoxInt( dec->params[ 0 ] );
+		break;
+	case STATUS_SUCCESS:
+		result = ( AINodeStatus_t ) AIUnBoxInt( dec->params[ 1 ] );
+		break;
+	case STATUS_RUNNING:
+		result = ( AINodeStatus_t ) AIUnBoxInt( dec->params[ 2 ] );
+		break;
+	}
+
+	return result;
```

Comment:
This should probably default to returning the original result if it is not one of the 3 valid values. Unless we are going to tighten up `decorator return` (and this one) to forbid random statuses.

Reply:
Did that.

If we really wanted to tighten up, we should probably do it when parsing. That would be rather easy, as we only allow constants as arguments to decorators and actions.